torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -358,14 +358,19 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
   // Don't need to hold lock while calling tensorpipe API.
   lock.unlock();
 
+  futureResponseMessage->addCallback([this]() {
+    // Decrease the callcount through a callback so it is only decremented once
+    // per future.
+    --clientActiveCalls_;
```

Comment:
Do we need a lock to protect this?

Reply:
It's an atomic, so we should be good without a lock.